Contributing-code-to-Oppia.md

```diff
@@ -65,21 +65,30 @@ Welcome! Please make sure to follow the instructions above if you haven't alread
 
 ### Choosing a good first issue
 
-When you've found a good first issue you'd like to tackle, please leave a comment on it that:
+When you've found a good first issue you'd like to tackle, please investigate it first to understand why the issue is happening. Here are some things you should do:
 
-- @-mentions the lead(s) of the corresponding project that the issue falls under (see details for [LaCE](https://github.com/orgs/oppia/projects/3/views/8?pane=info), [Contributor Dashboard](https://github.com/orgs/oppia/projects/18/views/4?pane=info) or [Developer Workflow](https://github.com/orgs/oppia/projects/8?pane=info)), letting them know you'd like to work on it.
-- says which part of the issue you're taking, if the issue is divided into subparts, and
-- describes in more detail how you'd tackle the issue (e.g. explain which file(s) you would modify and what changes you would make). If your explanation makes sense, we'll assign the issue to you.
+- Read the entire discussion thread to understand what has been tried so far.
+- Try to reproduce the issue on your local dev server. (For Contributor Dashboard issues, the [[Contributor Dashboard wiki page|Contributor-dashboard]] has some useful setup information.)
+- Figure out why the problem is happening, and find the relevant code in the Oppia repository to change. (The [['Finding the right code to change' wiki page|Find-the-right-code-to-change]] might be helpful.) If you have trouble with this, feel free to ask on [GitHub Discussion](https://github.com/oppia/oppia/discussions) and explain what you've tried doing so far.
+- If the issue is easy to fix, try to get a rough fix working on your local dev server!
 
-Once you've written that comment, feel free to go ahead and submit a PR for it by following the [[instructions for making a PR|Make-a-pull-request]]. You don't need to wait for approval to get started! If you run into any issues, feel free to create a [GitHub Discussion](https://github.com/oppia/oppia/discussions) and get help from the Oppia community, or [request a mentor](https://forms.gle/udsRP4WQgLcez9Zm8) if you'd like individual support.
+Once you have a good understanding of the issue, you can ask for it to be assigned to you by leaving a comment as follows:
+
+- Explain clearly how you'd tackle the issue (at a minimum, point to which file(s) you'd modify and describe the changes you'd make).
+- If possible, show a screenshot or code snippet demonstrating your proposed fix.
+- @-mention the leads of the corresponding project the issue falls under, letting them know you'd like to work on it. (The leads are: @SanjaySajuJacob and @Priyansh61 for [LaCE](https://github.com/orgs/oppia/projects/3/views/8?pane=info); @sagangwee and @chris7716 for [Contributor Dashboard](https://github.com/orgs/oppia/projects/18/views/4?pane=info); @U8NWXD and @gp201 for [Developer Workflow](https://github.com/orgs/oppia/projects/8?pane=info).)
```

Comment:
I worry that this will become out of date as leads change. How about we make sure leads are specified on the project boards and point people to that? This way there will be a single source of truth for who the leads are

Reply:
@U8NWXD That's what I did before -- it doesn't seem to have worked though, people keep @-mentioning me instead :) So I would suggest, let's try this and see if it works? If so then we can figure out how to make it more long-term-proof at a later stage.

Reply:
Sure. My other suggestion would be to indicate the person to contact in the issue, that way contributors don't have to look elsewhere. That's probably going to become out of date even faster though, so this might be a good compromise

Reply:
Yeah, agreed, and it'll be a lot of overhead to include on a per-issue basis. So yup, I'd suggest leaving it as-is for now and seeing how it goes.